edi_account/models/account_move.py

```diff
@@ -29,3 +46,9 @@ def action_invoice_paid(self):
         if self:
             self._event("on_paid_account_move").notify(self)
         return result
+
+    def action_generate_edi(self):
+        """This launches a notification for generating missing manual EDIs"""
+        self._event("on_generate_account_edi").notify(
```

Comment:
This is going to trigger the event even if you have not records.
Also what would be the behavior?
Maybe we can discuss how to do it generally...

Reply:
As discussed in call, we could have:
1. model_ids on exc type
2. enable_domain to filter records matching the type
3. enable_snippet here you could do what you want and return true/false when it applies or not to the current record

The next step would be: automatically show buttons depending on this based on a serialized computed field that tells you per exchange type which btn show.